### ISSUE TITLE
Get back the special measurements dropdown with the possibility of filter by category

### ISSUE THREAD

**Issue by xavijam** [CartoDB/cartodb#12386]
It was initially created [here](https://github.com/CartoDB/cartodb/issues/11618), and then changed again to a normal dropdown filtered in the client. So, finally, we have decided to get back to the first solution and use "backend filtering" using DO queries. That means we will need help in order to create the "search-as-you-type" functionality with backend requests. cc @CartoDB/builder-backend 

![screen shot 2017-02-23 at 10 25 01](https://cloud.githubusercontent.com/assets/132146/23253259/246054ea-f9b4-11e6-9d46-b8e114e944d1.png)

- Step 1 (measurament):
![screen shot 2017-02-23 at 10 25 29](https://cloud.githubusercontent.com/assets/132146/23253307/57adf2ee-f9b4-11e6-80f9-8b3888ffae1a.png)

- Step 2 (segment):
![screen shot 2017-02-23 at 10 25 33](https://cloud.githubusercontent.com/assets/132146/23253308/58002514-f9b4-11e6-95ee-027648eb570d.png)

**Comment by nobuti** [comment]
@ethervoid do you have the query to filter measurements using the search box above?

**Comment by nobuti** [comment]
@ethervoid @javitonino I'm afraid we will need a query to get a measurement by its id. The thing is when you enter in an analysis to edit it, in that moment you have only the measurement id. Because we are changing the way we handle measurements, requesting only those measurements that fit in the user's selection parameters (filter and search), we need a way to grab the information about the selected measurement. 

Until now what we were doing is request the whole measurement collection and look for that id in the frontend. But with these changes, it's likely the returned measurements don't include the selected.

I've been looking at the docs and I found [OBS_GetMeasureById](https://carto.com/docs/carto-engine/data/measures-functions#obsgetmeasurebyidgeomref-text-measureid-text-boundaryid-text) and [OBS_GetData](https://carto.com/docs/carto-engine/data/measures-functions#obsgetdataids-arraytext-metadata-json) but they need additional params beside the measurement id. Is there any simple way to get the metadata linked to a measurement id?

**Comment by juanignaciosl** [comment]
AFAIK we're gonna wrap the function so it _can_ be paged, but paging can be optional, isn't it? Adding an optional id filter as well shouldn't be hard. Am I right, @ethervoid ?

**Comment by ethervoid** [comment]
@buti with this query should be enough:
```sql
select numer_id, numer_name FROM cdb_dataservices_client.OBS_GetAvailableNumerators((select extent from _summary)) WHERE numer_id = '{{ numer_id }}';
```

**Comment by saleiva** [comment]
hi guys, there is something I've spotted while doing acceptance.
Simple options on any dropdown should be blue (for example in the region dropdown they are black now).

**Comment by xavijam** [comment]
In any design, dropdown list items appear with color black... is this new?

**Comment by saleiva** [comment]
@urbanphes can you take a look at what's going on?

**Comment by nobuti** [comment]
@saleiva @urbanphes open an issue for that: https://github.com/CartoDB/cartodb/issues/12426

**Comment by nobuti** [comment]
We found a problem related with selects editor: #12427. But the goal of this issue is achieved, we have measurements and filters up and running and fully async. So 🚀 .

**Comment by nobuti** [comment]
Closed by #12415.
